docs/src/main/asciidoc/security-keycloak-admin-client.adoc

```diff
@@ -0,0 +1,133 @@
+include::{includes}/devtools/extension-add.adoc[]
+
+This will add the following to your build file:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-keycloak-admin-client-reactive</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
+</dependency>
+----
+
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-keycloak-admin-client-reactive")
+implementation("io.quarkus:quarkus-resteasy-reactive-jackson")
+----
+
+We also are going to need a simple resource with a `Keycloak` injected as request scoped CDI bean.
+
+[source,java]
+----
+package org.acme.keycloak.admin.client;
+
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.representations.idm.RoleRepresentation;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import java.util.List;
+
+@Path("/api/admin")
+public class RolesResource {
+
+        @Inject
+        Keycloak keycloak;
+
+        @GET
+        @Path("/roles")
+        public List<RoleRepresentation> getRoles() {
+            return keycloak.realm("quarkus").roles().list();
+        }
+
+}
+----
+
+Now, you can produce the `Keycloak` bean yourself, or you can instruct Quarkus to produce `Keycloak` bean.
+
+NOTE: If you produce the bean yourself, please remember that `org.keycloak.admin.client.Keycloak` implements `AutoCloseable` and it's important to avoid resource exhaustion and related exceptions. If you configure your beans as explained below, Quarkus will do it for you.
+
+The Keycloak admin client needs an access token as its Admin REST API requires authorization.
+If you exchange user's credentials for the access token, here is an example configuration for the `password` grant type:
+
+[source,properties]
+----
+quarkus.keycloak.admin-client=true
+quarkus.keycloak.admin-client.server-url=http://localhost:8081
+quarkus.keycloak.admin-client.realm=quarkus
+quarkus.keycloak.admin-client.client-id=quarkus-client
+quarkus.keycloak.admin-client.username=alice
+quarkus.keycloak.admin-client.password=alice
+quarkus.keycloak.admin-client.grant-type=PASSWORD <1>
+----
+<1> Use `password` grant type.
+
+An example using the `client-credentials` grant type needs only a minor adjustments:
+
+[source,properties]
+----
+quarkus.keycloak.admin-client=true
+quarkus.keycloak.admin-client.server-url=http://localhost:8081
+quarkus.keycloak.admin-client.realm=quarkus
+quarkus.keycloak.admin-client.client-id=quarkus-client
+quarkus.keycloak.admin-client.client-secret=secret
+quarkus.keycloak.admin-client.username= # remove default username
+quarkus.keycloak.admin-client.password= # remove default password
+quarkus.keycloak.admin-client.grant-type=CLIENT_CREDENTIALS <1>
```

Comment:
Hi @michalvavrik It all looks good to me now, thanks, one last thing I'd like to suggest. I believe, to avoid the confusion (`quarkus-oidc-client` is designed to request the tokens in a provider API neutral way), it makes sense to add a note somewhere under this example, which is showing how to get a token using a client cred grant, to add a note like "Note that the oidc-client can also be used to acquire tokens" and link to the guide, something like that

Reply:
Sure, done. I know you can use the Keycloak admin client to get a token, but I thought (expected) the other way around (you can use OIDC client to get a token that you set when you are creating the admin client). Thank you.